### ISSUE TITLE
Can't create a memberExpression with property type of 'Literal'

### ISSUE THREAD

**Issue by juliankrispel** [benjamn/recast#235]
I'm trying to generate ast types programmatically and print out the code.

You can [see this here](http://astexplorer.net/#/mNvVofHYVs).

This is how I'm trying to generate the `MemberExpression`

```js
builder.memberExpression(builder.identifier('boo'), builder.literal('boom'));
```

The output I get is this:

```js
boo."boom"
```

Looks like a bug right? Or am I doing something wrong?

**Comment by benjamn** [comment]
There used to be a warning about this, but you need to pass a third boolean argument for the `.computed` property of the `MemberExpression`. The default is `false` (i.e. not computed, a typical `.identifier` property), but I think what you want is
```js
builder.memberExpression(builder.identifier('boo'), builder.literal('boom'), true)
```

**Comment by juliankrispel** [comment]
@benjamn I was just about to say that I'd found it by having a closer look at the MemberExpression implementation. Couldn't the computed flag just be turned on automatically if the property is a literal? Otherwise what you get is an illegal expression. I'd be happy to help implement this and write a test for it. I'm guessing this issue would be better placed in the ast-types repo right?

**Comment by benjamn** [comment]
I like that idea! I think you would need to change the `defaults["false"]` argument [here](https://github.com/benjamn/ast-types/blob/86491e9e1b5421a9ec4d9ebccd3422b8f0cf29d6/def/core.js#L308) to a function that examines `this.property` and decides whether it's an `Identifier` or not. Just below that line, you can find an example of a custom default function for for `RegExp`-valued `Literal` nodes: https://github.com/benjamn/ast-types/blob/86491e9e1b5421a9ec4d9ebccd3422b8f0cf29d6/def/core.js#L332-L347

If you have time to submit an `ast-types` PR, great. If not, I'll try to get around to it soon.

**Comment by juliankrispel** [comment]
@benjamn thanks! Doing it!

**Comment by juliankrispel** [comment]
@benjamn https://github.com/benjamn/ast-types/pull/135

**Comment by juliankrispel** [comment]
fixed in benjamn/ast-types#135
